staging/src/k8s.io/apimachinery/pkg/util/proxy/dial.go

```diff
@@ -101,6 +100,21 @@ func dialURL(ctx context.Context, url *url.URL, transport http.RoundTripper) (ne
 				return nil, err
 			}
 
```

Comment:
according to https://github.com/kubernetes/kubernetes/pull/109750, Handshake validates host... why is this still needed for this path?

Reply:
also, should this switch to HandshakeContext (and tls.Dial below switch to tls.Dialer#DialContext?)

Reply:
Follow up https://github.com/kubernetes/kubernetes/pull/110079